src/Components/WebAssembly/WebAssembly/src/HotReload/WebAssemblyHotReload.cs

```diff
@@ -27,18 +25,20 @@ public static class WebAssemblyHotReload
 
         internal static async Task InitializeAsync()
         {
-            // Determine if we're running under a hot reload environment (e.g. dotnet-watch).
-            // It's insufficient to know it the app can be hot reloaded (HotReloadFeature.IsSupported),
-            // since the hot-reload agent might be unavailable.
-            if (Environment.GetEnvironmentVariable("DOTNET_MODIFIABLE_ASSEMBLIES") != "debug")
-            {
-                return;
-            }
-
             _hotReloadAgent = new HotReloadAgent(m => Debug.WriteLine(m));
 
-            var jsObjectReference = (IJSUnmarshalledObjectReference)(await DefaultWebAssemblyJSRuntime.Instance.InvokeAsync<IJSObjectReference>("import", "./_framework/blazor-hotreload.js"));
-            await jsObjectReference.InvokeUnmarshalled<Task<int>>("receiveHotReload");
+            try
+            {
+                // Attempt to read previously applied hot reload deltas. dotnet-watch and VS will serve the script that can provide results from local-storage and
+                // the injected middleware if present.
+                var jsObjectReference = (IJSUnmarshalledObjectReference)(await DefaultWebAssemblyJSRuntime.Instance.InvokeAsync<IJSObjectReference>("import", "./_framework/blazor-hotreload.js"));
+                await jsObjectReference.InvokeUnmarshalled<Task<int>>("receiveHotReload");
+            }
+            catch (Exception ex)
+            {
+                // The middleware may be unavailable in some cases - https://github.com/dotnet/aspnetcore/issues/34962. This is not a reason to fail, so swallow the exception and continue.
```

Comment:
The description in the linked issue doesn't say why the middleware might be missing. The closest it comes is this:

> WebAssemblyHotReload should also be robust against a missing browser refresh middleware

Am I missing something? Could you update this comment to give the reason why the middleware could be missing? Is it just something that the application author gets to control?

Reply:
I'm asking because we don't want to be in a situation where things are failing and we just log but carry on, as this ends up accumulating warnings that developers have to ignore and worsen the overall experience.

If it's legit for the middleware not to be there, it seems like we shouldn't be logging warnings. And if it's not legit, it seems like we shouldn't be suppressing it.

Reply:
> Could you update this comment to give the reason why the middleware could be missing? 

The signal we use to determine if hot reload is enabled is the `MetadataUpdater.IsSupported` flag which looks for the environment variables that are configured by ctrl-f5 / dotnet-watch or if the app is being debugged. Currently, VS does not support hot reload for WASM during debugging and does not inject the middleware at that time which causes the error. 

An alternative is to disallow this code when `Debugger.IsAttached` is true. But it would mean we would have to patch the runtime to enable hot reload if VS enables it in a point release. That said, I hadn't thought about the 404 (that @javiercn pointed out) or the warnings.

Reply:
@javiercn thinks we might be able to something using SWA. Removed this for now